packages/providers/src/provider.ts

```diff
@@ -209,15 +211,34 @@ export default class Provider {
   operations: ReturnType<typeof getOperationsSdk>;
   cache?: MemoryCache;
 
+  consensusParamsCache?: ConsensusParameters;
+
   constructor(
     /** GraphQL endpoint of the Fuel node */
     public url: string,
     public options: ProviderOptions = {}
   ) {
+    // TODO: Disable or discourage the usage of this constructor in favour of connect()
     this.operations = this.createOperations(url, options);
     this.cache = options.cacheUtxo ? new MemoryCache(options.cacheUtxo) : undefined;
   }
 
+  static async connect(url: string, options: ProviderOptions = {}) {
+    const provider = new Provider(url, options);
+    const { consensusParameters } = await provider.getChain();
+    provider.consensusParamsCache = consensusParameters;
+    return provider;
+  }
+
+  async invalidateConsensusParamsCache() {
+    const { consensusParameters } = await this.getChain();
+    this.consensusParamsCache = consensusParameters;
+  }
+
+  updateUrl(url: string) {
+    this.operations = this.createOperations(url);
+  }
```

Comment:
We should probably invalidate the consensus params cache when we update the URL. I'm noting down this as a TODO.